cluster/src/main/java/io/snappydata/gemxd/MemoryNotificationFactory.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2016 SnappyData, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+package io.snappydata.gemxd;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by sachin on 24/8/16.
+ */
+public class MemoryNotificationFactory {
+  private static List<MemoryNotificationListener> notificationListeners = new ArrayList<MemoryNotificationListener>();
+
+
+  public static void attachListener(MemoryNotificationListener listener) {
+    notificationListeners.add(listener);
+  }
+
+
+  public static void notifyCriticalUp() {
+    for (MemoryNotificationListener mnl : notificationListeners) {
+      mnl.notifyCriticalUp();
+    }
+  }
+
+  public static void notifyEvictionUp() {
+    for (MemoryNotificationListener mnl : notificationListeners) {
+      mnl.notifyEvictionUp();
+    }
+  }
+
+
+}
```

Comment:
See comment below. Both this class and LeadNodeMemoryListener looks redundant to me when ResourceListener can be used directly.